### PR TITLE
1283 survey update

### PR DESCRIPTION
**Pull request by willdoran**
This pull request makes the following changes:
- Adds is_public field to task
- Adds description to task
- Adds type to task

Test these changes by:
- Test via [301](https://github.com/ushahidi/platform-client/pull/301)

Fixes ushahidi/platform#1283

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform/1332)

### CODE REVIEW

**Comment by rjmackay** [comment]
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[migrations/20160803210356_add_type_to_stage.php, line 27 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform/1332#-KOlFclVgQC5lozUKX2_:-KOlFclWl7GEjt4vQzth:b-xq5kqm) ([raw file](https://github.com/ushahidi/platform/blob/cf492952c9f8b39738b6c3bb6c42065ef27935a1/migrations/20160803210356_add_type_to_stage.php#L27)):*
> ```PHP
>       $this->table('form_stages')
>         ->addColumn('type', 'string', [
>             'null' => true,
> ```

This probably shouldn't ever be null. Maybe default to `type = task` or `type = post` too ??

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform/1332)*

**Comment by rjmackay** [comment]
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[migrations/20160803210356_add_type_to_stage.php, line 27 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform/1332#-KOlFclVgQC5lozUKX2_:-KOlFkvYbx68AxVhYqod:b-c8hspz) ([raw file](https://github.com/ushahidi/platform/blob/cf492952c9f8b39738b6c3bb6c42065ef27935a1/migrations/20160803210356_add_type_to_stage.php#L27)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> This probably shouldn't ever be null. Maybe default to `type = task` or `type = post` too ??
</details>
Also: how do we migrate existing stages? first stage should become post.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform/1332)*

File: src/Core/Entity/FormStage.php [hunk 1]
```diff
@@ -20,18 +20,24 @@ class FormStage extends StaticEntity
...
+	protected $description;
+	protected $is_public;
 
 	// DataTransformer
 	protected function getDefinition()
 	{
 		return [
 			'id'       => 'int',
+			'description' => 'string',
+			'type'     => 'string',
```
Comment (rjmackay) [inline src/Core/Entity/FormStage.php]: You need to add validation for this field. Maybe the others too?

**Comment by willdoran** [comment]
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[migrations/20160803210356_add_type_to_stage.php, line 27 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform/1332#-KOlFclVgQC5lozUKX2_:-KOpkAPT4HNS3G6MAsXY:b-yhwb8h) ([raw file](https://github.com/ushahidi/platform/blob/cf492952c9f8b39738b6c3bb6c42065ef27935a1/migrations/20160803210356_add_type_to_stage.php#L27)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Also: how do we migrate existing stages? first stage should become post.
</details>
Ah yeah, I meant to discuss that yesterday, for the client, it assumes surveys without "post" tasks just behave as before with the tasks placed in tabs.

We would have to migrate all the old surveys otherwise and that might not be a great idea.

Not sure, this might need more discussion.

---

*[src/Core/Entity/FormStage.php, line 34 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform/1332#-KOk2D7tgW7Alt3GLec9-r1-34:-KOpkUSt-zV5dEWFmUKV:b-896fix) ([raw file](https://github.com/ushahidi/platform/blob/cf492952c9f8b39738b6c3bb6c42065ef27935a1/src/Core/Entity/FormStage.php#L34)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> You need to add validation for this field. Maybe the others too?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform/1332)*

**Pull request merged**